server/osc_dash.py

```
@@ -763,6 +763,54 @@ async def api_upload_chunk(
+        return JSONResponse(status_code=400, content={"error": "invalid filename"})
+    if not (filename.endswith(".jsonl") or filename.endswith(".jsonl.gz") or filename.endswith(".gz")):
+        return JSONResponse(status_code=400, content={"error": "file must be .jsonl or .gz"})
+
+    target_file = (TICKS_DIR / filename).resolve()
+    if not str(target_file).startswith(str(TICKS_DIR.resolve())):
+        return JSONResponse(status_code=400, content={"error": "invalid target path"})
+
+    lock = _get_target_lock(filename)
+    tmp_target = target_file.with_suffix(target_file.suffix + f".tmp_{time.time_ns()}")
+    try:
+        with open(tmp_target, "wb") as out_f:
```

Ruff (0.16.3)

[warning] Async functions should not open files with blocking methods like `open`

(ASYNC230)